### DISCUSSION TITLE
Блокировка обновления вычислений в полях

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#322] [Как это сделать?] [answered]
Добрый день!
Как можно заблокировать изменение в таблице, в том числе вычисляемых значений?
Например. 
Есть некий просчет. 
Когда он готов, он фиксируется. Блок таблицы блокирует введение и изменение ручных данных.
Но данные, которые зависят от внешних источников (например курса и т.п.) продолжают пересчитываться.

**Comment by juramebel** [comment]
Наверное придется вставить скрытое поле true/false и вставлять проверку в каждое вычисляемое поле? Плюс блочить ручное изменение таблиц..

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Глобально есть 2 подхода.

1. Это при создании берем без изменения и потом принудительно обновляем:
    - Значение берем с «Исполнять кодом только при добавлении»
    - Обновляем выполняя `recalculate` с параметром `field` (в этом случае код выполнится в момент рекалка)

2. Блокируем при постановке какого-то параметра. Тогда Код в поле будет выглядеть так:
    ```
    =: if(condition: #param = true; then: $#onfv; else: $calc)
    calc: ...
    
    // Те если контрольный параметр true то берем предыдущее значение этого поля, если нет, то считаем
    ```
